### DISCUSSION TITLE
String Obfuscation

### DISCUSSION THREAD

**Discussion by doctormeeee** [HemulGM/ChatGPT#53] [Q&A] [answered]
Hey there, I wonder if you tried to obfuscate constant strings such as the URL to OpenAI API when composing the APK file for publishing?

**Comment by HemulGM** [comment]
> Hey there, I wonder if you tried to obfuscate constant strings such as the URL to OpenAI API when composing the APK file for publishing?

No. Is this necessary?

**Reply by doctormeeee** [reply]
I guess no. I saw an URL to OpenAI in your source code. But when I analyzed your published apk, that particular URL seems to be hidden. So I wonder whether that could be due to any potential string obfuscation mechanism used when your build your source code as an apk? Otherwise, what could cause that OpenAI URL to be hidden in the apk though.

**Reply by HemulGM** [reply]
> I guess no. I saw an URL to OpenAI in your source code. But when I analyzed your published apk, that particular URL seems to be hidden. So I wonder whether that could be due to any potential string obfuscation mechanism used when your build your source code as an apk? Otherwise, what could cause that OpenAI URL to be hidden in the apk though.

You need to analyze the file libChatGPT.so inside apk file

**Reply by HemulGM** [reply]
![image](https://github.com/HemulGM/ChatGPT/assets/22197242/2a816457-c44c-4b5d-a25c-9fe8342ad6f4)

**Reply by doctormeeee** [reply]
Thanks so much for your reply! I will have a look at that .so file. I wonder if you used any vscode extension to view that so file? Or did you specify a particular editor for opening so files that the binary code can be automatically decoded? Thanks.

**Reply by HemulGM** [reply, marked as answer]
Extention HEX Editor

**Reply by AsadUllah1972** [reply]
Hi

**Reply by HemulGM** [reply]
hi
